src/react/useQueries.ts

```diff
@@ -0,0 +1,137 @@
+import * as React from 'react';
+import { Monoid } from 'fp-ts/lib/Monoid';
+import { Option, fromNullable } from 'fp-ts/lib/Option';
+import { QueryResult } from '../QueryResult';
+import { product, ObservableQuery } from '../Query';
+import {
+  EnforceNonEmptyRecord,
+  ObservableQueries,
+  ProductL,
+  ProductP,
+  ProductA,
+  VoidInputObservableQueries
+} from '../util';
+import { defaultMonoidResult } from './util';
+import { observable } from '../Observable';
+import { observeShallow } from '../observe';
+import { setoidShallow } from '../Strategy';
+
+function usePrevious<T>(value: T): Option<T> {
+  const ref = React.useRef<T>();
+  React.useEffect(() => {
+    ref.current = value;
+  });
+  return fromNullable(ref.current);
+}
+
+/**
+ * A React hook to observe an `ObservableQuery`
+ *
+ * @param query an `ObservableQueries`
+ * @param params input values for the query
+ * @param resultMonoid an optional monoid used to aggregate `QueryResult`s
+ * @returns the latest `QueryResult` to operate upon
+ *
+ * @example
+ * useQuery(randomQuery, randomQueryParams).fold(
+ *   () => "load",
+ *   () => "failed",
+ *   (randomQueryResult) => randomQueryResult
+ * )
+ */
+export function useQuery<A extends void, L, P>(
+  query: ObservableQuery<A, L, P>,
+  params?: A,
+  resultMonoid?: Monoid<QueryResult<L, P>>
+): QueryResult<L, P>;
+export function useQuery<A, L, P>(
+  query: ObservableQuery<A, L, P>,
+  params: A,
+  resultMonoid?: Monoid<QueryResult<L, P>>
+): QueryResult<L, P>;
+export function useQuery<A, L, P>(
+  query: ObservableQuery<A, L, P>,
+  params: A,
+  resultMonoid?: Monoid<QueryResult<L, P>>
+): QueryResult<L, P> {
+  const _resultMonoid = resultMonoid || defaultMonoidResult<L, P>();
+
+  const [state, setState] = React.useState<QueryResult<L, P>>(
+    _resultMonoid.empty
+  );
+
+  const previousInput = usePrevious(params);
+  const [inputEquality, setInputEquality] = React.useState(0);
+
+  React.useEffect(() => {
+    const inputChanged = previousInput.fold(
+      false,
+      previousInput => !query.inputSetoid.equals(previousInput, params)
+    );
+    if (inputChanged) {
+      setInputEquality(inputEquality + 1);
+    }
+  });
```

Comment:
no dependencies for this `useEffect`? Shouldn't they be

```suggestion
  }, [previousInput, inputEquality);
```

?

Reply:
not sure, but I wouldn't add:
- `inputEquality` because this effect is meant to update it, it'd be a circular dependency of some sort?
- `previousInput` because it's an `Option<>` and changes ref at every render (unless we memoize it somehow..). But in the end this `previousInput` changing is already the check contained inside of `useEffect`, which in the current version always re-executes (having no deps). So at best this would be a perf optimization I think